### PR TITLE
feat: add proto definitions necessary to support generic PR's

### LINKED ISSUE #5: Review generic data structure to represent a pull request
## Description
Currently, we are tightly coupled with githubs presentation of what a pull request is which will make it hard to support other providers.

### Motivation
Closing this issue will get us one step closer to support alternative codehost providers such as gitlab and bitbucket.

## Proposed solution
Define our presentation of what a pull request is.

### PR DESCRIPTION
**Pull request by zolamk**
## Description
Adds proto definitions necessary to support generic PR's

## Related issue

Closes #5 

## Type of change

**New feature** (non-breaking change which adds functionality)

## How was this tested?
TBD

## Checklist

- [x] I have performed a self-review of my code
- [ ] I have ran `task check -f` and have no issues

## Code review and merge strategy (ship/show/ask) 

**Ask**: this pull request requires a code review before merge

### CODE REVIEW

File: pb/entities/git_repo.proto [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    int64 external_created_at = 7;
+    int64 external_updated_at = 8;
+    bool is_public = 9;
+    int32 reviews_count = 10;
+    int32 branches_count = 11;
+    int32 commits_count = 12;
+    repeated MergeMethodOptions allowed_merge_methods = 13;
+    repeated DeleteHeadOption allowed_delete_head_options = 14;
+    DeleteHeadOption default_delete_head_option = 15;
+    bool always_with_merge_commit = 16;
```
Comment (zolamk) [inline pb/entities/git_repo.proto:30]: we should only keep the fields we are using

File: pb/entities/label.proto [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+syntax = "proto3";
+
+package entities;
+
+option go_package = "github.com/reviewpad/api/go/entities";
+
+message Label {
+    string id = 1;
+    string external_id = 2;
```
Comment (zolamk) [inline pb/entities/label.proto:13]: no need for external id

File: pb/services/hosts.proto [hunk 3]
```diff
@@ -8,12 +8,14 @@ package services;
...
 import "entities/code_host.proto";
 import "entities/comments.proto";
+import "entities/code_review.proto";
 
 option go_package = "github.com/reviewpad/api/go/services";
 
 service Hosts {
   // Comments
   rpc PostGeneralComment(PostGeneralCommentRequest) returns (PostGeneralCommentReply) {}
+  rpc GetCodeReview(GetCodeReviewRequest) returns (GetCodeReviewReply) {}
```
Comment (zolamk) [inline pb/services/hosts.proto:18]: should be called `GetPullRequest`

File: pb/services/hosts.proto [hunk 4]
```diff
@@ -49,3 +51,16 @@ message GetApiDistributionReply {
   map<string, ApiRequestDistribution> distribution = 2;
   int32 total_cost = 3;
 }
+
+message GetCodeReviewRequest {
+  string slug = 1;
+  string external_repo_id = 2;
+  entities.Host host = 3;
+  string host_uri = 4;
+  int32 review_number = 5;
```
Comment (zolamk) [inline pb/services/hosts.proto:60]: we should rename this to `number`

File: pb/services/hosts.proto [hunk 5]
```diff
@@ -49,3 +51,16 @@ message GetApiDistributionReply {
   map<string, ApiRequestDistribution> distribution = 2;
   int32 total_cost = 3;
 }
+
+message GetCodeReviewRequest {
```
Comment (zolamk) [inline pb/services/hosts.proto:55]: should be named `GetPullRequestRequest`

File: pb/services/hosts.proto (hunk 3 above)
Comment (zolamk) [inline pb/services/hosts.proto:18]: we need a new grpc call to get commit files separetly

**Review by marcelosousa — APPROVED** [review]
Mob reviewed with @ferreiratiago and @shay2025

**Review by ferreiratiago — APPROVED** [review]

**Pull request merged**